joinmarket-webui/jam
# dev(regtest): prepare 'wait-for-bitcoind' feature

This is the test branch for https://github.com/joinmarket-webui/jam-docker/pull/76.

Once https://github.com/joinmarket-webui/jam-docker/pull/76 is merged and a new `jam-dev-standalone:master` image has been created, the last commit of this PR will be reverted and this PR can be undrafted.

Please do not merge it without reverting the image base for the additional contains from `jam-dev-standalone:wait-for-bitcoind` back to `jam-dev-standalone:master`!

This PR removes the need for the `READY_FILE` in the secondary JM instance (`jm_regtest_joinmarket2`) and relies on the `WAIT_FOR_BITCOIND` env var to default to `true`. The tertiary instance (`jm_regtest_joinmarket3`) verifies that `WAIT_FOR_BITCOIND` can be disabled and the `READY_FILE` feature continues to work as expected. Please keep in mind that the primary test instance (`jm_regtest_joinmarket`) does not use the official "dev"-base image at all!

## How to test
Build your env environment with `npm run regtest:build` (makes sure new images are pulled) and run `npm run regtest:up`.
The environment should start normally as usual. `npm run regtest:init` should be able to create wallets on all instances.

You should see the following in the logs:

For `jm_regtest_joinmarket3` (enabling `WAIT_FOR_BITCOIND` and not using `READY_FILE`):
```
jm_regtest_joinmarket2          | Waiting for bitcoind to accept RPC requests...
jm_regtest_bitcoind             | 2022-11-02T10:36:08.423004Z ThreadRPCServer method=getblockchaininfo user=joinmarket2
jm_regtest_joinmarket2          | Successfully waited for bitcoind to accept RPC requests.
```

For `jm_regtest_joinmarket3` (disabling `WAIT_FOR_BITCOIND` and using `READY_FILE`):
```
jm_regtest_joinmarket3          | Waiting for file /root/.regtest-initializer/btc_fully_synched to be created...
jm_regtest_joinmarket3          | Successfully waited for file /root/.regtest-initializer/btc_fully_synched to be created.
```
You should NOT see `Waiting for bitcoind to accept RPC requests... for `jm_regtest_joinmarket3`.
